pySDC/implementations/problem_classes/AdvectionEquation_ND_FD.py

```diff
@@ -48,91 +47,87 @@ def __init__(
             liniter (int): Max. iterations for GMRES
             direct_solver (bool): Whether to solve directly or use GMRES
             bc (str): Boundary conditions
-            ndim (int): Number of dimensions. Is set automatically if left at None.
         """
 
-        # make sure parameters have the correct form
-        if not (type(nvars) is tuple and type(freq) is tuple) and not (type(nvars) is int and type(freq) is int):
-            print(nvars, freq)
-            raise ProblemError('Type of nvars and freq must be both either int or both tuple')
+        # make sure parameters have the correct types
+        if not type(nvars) in [int, tuple]:
+            raise ProblemError('nvars should be either tuple or int')
+        if not type(freq) in [int, tuple]:
+            raise ProblemError('freq should be either tuple or int')
 
-        if ndim is None:
-            if type(nvars) is int:
-                ndim = 1
-            elif type(nvars) is tuple:
-                ndim = len(nvars)
+        # transforms nvars into a tuple
+        if type(nvars) is int:
+            nvars = (nvars,)
 
+        # automatically determine ndim from nvars
+        ndim = len(nvars)
         if ndim > 3:
             raise ProblemError(f'can work with up to three dimensions, got {ndim}')
 
-        if type(freq) is tuple:
-            for f in freq:
-                if f % 2 != 0 and bc == 'periodic':
-                    raise ProblemError('need even number of frequencies due to periodic BCs')
-        else:
-            if freq % 2 != 0 and freq != -1 and bc == 'periodic':
```

Comment:
I supposed that the `freq != -1` was a residue from a previous implementation, since you can avoid this condition using `nvars=(4,)` and `freq=(-1,)`, which would not work. Or is there something I did not realize @pancetta ?

Reply:
No actually I implemented `freq=-1` as a Gaussian. But only for 1D, and periodic, so it should fail when you supply a tuple with more than one entry, I forgot that you can also supply `freq=(-1)`. I guess it's not the cleanest implementation...

Reply:
Do we want to keep that option ?

Reply:
I want to keep the Gaussian as an option, but I am open to changing the interface.

Reply:
Ok let me see what I can do. Just to be clear, Gaussian works only with periodic boundary condition in 1D ?

Reply:
It only works in 1D. If you just use it for initial conditions than ofc the boundary conditions don't matter, but the exact solution at later times only works with periodic boundaries. Also, the domain is hardcoded from 0 to 1 [here](https://github.com/Parallel-in-Time/pySDC/blob/1b0d8989584f9b1983b64d003ca6ad208cb5d6a4/pySDC/implementations/problem_classes/AdvectionEquation_ND_FD.py#L188)

Reply:
Ok thanks. But the domain was already hardcoded to [0,1] with the generation of `xvalues`, so that's not an issue :wink:

Reply:
@brownbaerchen what's the default value you want to use for `sigma` btw ?

Reply:
I used `simga=6e-2` at some point. But we can take care of that later. I guess that should be decided together with the spatial resolution